src/Functions/functions.js

```diff
@@ -0,0 +1,57 @@
+import firebase from "../Components/firebase";
+
+export const handleSubmit = (
+  { Email, setEmail },
+  { Password, setPassword },
+  { Username, setUsername }
+) => {
+  firebase
+    .auth()
+    .createUserWithEmailAndPassword(Email, Password)
+    .then((authUser) => {
+      alert("User Created");
+      authUser.user.updateProfile({
+        displayName: Username,
+      });
+    })
+    .catch((e) => {
+      console.log(e);
+      alert("Error Occured or user not created");
+    });
+
+  setEmail("");
+  setPassword("");
+  setUsername("");
+};
+
+export function CheckStatus({ user, setUser }) {
```

Comment:
CheckStatus should not be here. Better way to check if user is logged in is to create a `getUser` function which returns `auth().currentUser`, if this returns falsy it means user is logged out. Do this at component level